test/com/linkedin/parseq/TestTasks.java

```diff
@@ -89,6 +91,59 @@ public void onResolved(Promise<String> stringPromise) {
     assertEquals(Boolean.TRUE, resultRef.get());
   }
 
+  @Test
+  public void testFromFuture() {
+    String result = "FROMFUTURERESULT";
+    Task<String> task = Task.fromFuture(() ->{
+      CompletableFuture<String> completableFuture
+          = CompletableFuture.supplyAsync(() -> result);
+      return completableFuture;
+    });
+    String taskResult = runAndWait("testToTask.success", task);
+    assertEquals(result, taskResult);
+  }
+
+  @Test
+  public void testFromFutureWithTimeConsumingFuture() throws InterruptedException {
+    String result = "FROMFUTURERESULT";
+    Task<String> task = Task.fromFuture(() ->{
+      CompletableFuture<String> completableFuture
+          = CompletableFuture.supplyAsync(() -> {
+        try {
+          Thread.sleep(1000);
```

Comment:
Please decrease it to something smaller e.g. 100ms. We don't want one unit test to take 1s.

Reply:
Fixed